test/controllers/project_controller_test.exs

```diff
@@ -106,142 +68,66 @@ defmodule CodeCorps.ProjectControllerTest do
-    @tag :authenticated
+    @tag authenticated: :admin
     test "creates and renders resource when attributes are valid", %{conn: conn, current_user: current_user} do
       organization = insert(:organization)
       insert(:organization_membership, role: "admin", member: current_user, organization: organization)
-
-      payload =
-        build_payload
-        |> put_attributes(@valid_attrs)
-        |> put_relationships(organization)
-
-      path = conn |> project_path(:create)
-
-      json = conn |> post(path, payload) |> json_response(201)
-
-      id = json["data"]["id"]
-      assert id
-      project = Project |> Repo.get(id)
-
-      assert project
-      assert project.title == "Valid project"
-      assert project.description == "Valid project description"
-      assert project.long_description_markdown == "Valid **markdown**"
-      assert project.long_description_body == "<p>Valid <strong>markdown</strong></p>\n"
-      assert project.organization_id == organization.id
+      attrs = @valid_attrs |> Map.merge(%{organization: organization})
+      assert conn |> request_create(attrs) |> json_response(201)
     end
 
-    @tag :authenticated
-    test "does not create resource and renders errors when attributes are invalid", %{conn: conn, current_user: current_user} do
-      organization = insert(:organization)
-      insert(:organization_membership, role: "admin", member: current_user, organization: organization)
-
-      payload =
-        build_payload
-        |> put_attributes(@invalid_attrs)
-        |> put_relationships(organization)
-
-      path = conn |> project_path(:create)
-
-      errors = conn |> post(path, payload) |> json_response(422) |> Map.get("errors")
-
-      assert errors != %{}
+    @tag authenticated: :admin
+    test "renders 422 when attributes are invalid", %{conn: conn} do
+      assert conn |> request_create(@invalid_attrs) |> json_response(422)
     end
 
-    test "does not create resource and renders 401 when unauthenticated", %{conn: conn} do
-      path = conn |> project_path(:create)
-      assert conn |> post(path) |> json_response(401)
+    test "renders 401 when unauthenticated", %{conn: conn} do
+      assert conn |> request_create |> json_response(401)
     end
 
     @tag :authenticated
-    test "does not create resource and renders 403 when not authorized", %{conn: conn, current_user: current_user} do
+    test "renders 403 when not authorized", %{conn: conn} do
       organization = insert(:organization)
-      insert(:organization_membership, role: "contributor", member: current_user, organization: organization)
-
-      payload =
-        build_payload
-        |> put_attributes(@valid_attrs)
-        |> put_relationships(organization)
-
-      path = conn |> project_path(:create)
-      assert conn |> post(path, payload) |> json_response(403)
+      attrs = @invalid_attrs |> Map.merge(%{organization: organization})
+      assert conn |> request_create(attrs) |> json_response(403)
```

Comment:
I was having trouble getting this test to pass and this was the only solution that worked. I couldn't figure out why the usual request_create wasn't working so I just included the attributes. I'll look at it again today.